src/components/Onboarding/ColumnSelectorSidebar.tsx

```diff
@@ -146,6 +147,22 @@ export default function ColumnSelectorSidebar(props: { plotArgs: Args }) {
   // Create charts whenever the column selection changes
   useEffect(recommendCharts, [selectedColumns]);
 
+  /**
+   * This function will move aggregate to transform to add groupby operation
+   * @param spec current graph spec
+   */
+  function handleAggregate(spec: GraphSpec): GraphSpec {
+    return produce(spec, (gs) => {
+      if ('aggregate' in gs.encoding.x && !('field' in gs.encoding.x)) {
+        (gs.encoding.x as any).type = 'nominal';
+        (gs.encoding.x as any).field = (gs.encoding?.y as any).field;
+      } else if ('aggregate' in gs.encoding.y && !('field' in gs.encoding.y)) {
+        (gs.encoding.y as any).type = 'nominal';
+        (gs.encoding.y as any).field = (gs.encoding.x as any).field;
+      }
+    });
+  }
```

Comment:
Not sure I understand why we are flip flopping the axes here... I think we can just apply the count as a transformation. In the Vega docs, it says that a count on one field applies to all fields, so you can use that to your advantage to assign count to a hidden filter.

Reply:
Removed this